### ISSUE TITLE
Non-uniqueness of interpolated parameters

### ISSUE THREAD

**Issue by mattwthompson** [openforcefield/openff-interchange#31]
In SMIRNOFF world, given a topology and force field that have already been through the typing step, I've been thinking that there's a 1:1:1 mapping between topology objects (i.e. tuple representing a `Bond`), SMIRNOFF parameters (i.e. `BondType`), and parametrized data (i.e. some `Potential` object, or `openmm.Force` object). This chain of key-val pairs is not so clear for interpolated parameters (currently proper torsions and harmonic bonds), in which one SMIRNOFF parameter maps onto arbitrarily many parametrized data.

For example, here are a couple lines for a test force field used in the toolkit:

```XML
     <Bond smirks="[#6X4:1]-[#6X3:2]" id="b2" k="634.0 * kilocalories_per_mole/angstrom**2" length="1.51 * angstrom     "/>
     <Bond smirks="[#6X3:1]-[#6X3:2]" id="b3" k_bondorder1="100.0*kilocalories_per_mole/angstrom**2" k_bondorder2="4000.0*kilocalories_per_mole/angstrom**2" length="1.5 * angstrom"/>
```

If 100 bonds in a topology are assigned to have bond `b2`, they will always be representing by those two physical quantities, no other information needed, and always the same data. But if a collection of 100 bonds are assigned bond `b3`, each bond may have slightly lengths (and, often, force constants) depending on the partial bond order found in the topology and/or used in the parametrization process.

Some ideas that come to mind

* Enforce uniqueness by replacing the SMIRKS key with a key that tacks on the bond order, data from which the  parametrized potential could be re-constructed.
* Bypass de-duplication altogether

**Comment by mattwthompson** [comment]
A solution to #32 may pick this up for free

**Comment by mattwthompson** [comment]
I believe this was resolved in #137
